pkg/master/controller.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package master
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/endpoint"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/namespace"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/service"
+	servicecontroller "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/service/allocator/controller"
```

Comment:
A while back said we want to have more functionally oriented directories, i.e. pkg/service/controller/*.

Cloud Provider is probably a special case, because external load balancers can only exist (today) with a cloud provider, whereas this does not depend on them.  

----- Original Message -----
> > +*/
> > +
> > +package master
> > +
> > +import (
> > +	"fmt"
> > +	"net"
> > +	"reflect"
> > +	"time"
> > +
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/endpoint"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/namespace"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/service"
> > +	servicecontroller
> > "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/service/allocator/controller"
> 
> This naming is a little awkward given the existence of
> pkg/cloudprovider/servicecontroller, but not worth worrying too much about.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7704/files#r29631092